apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,85 @@
+/// <para>The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when executing the
+/// partition object.For more information see
+/// [ReadRequest DataBoostEnabled](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.ReadRequest#Google_Cloud_Spanner_V1_ReadRequest_DataBoostEnabled) and
+/// [ExecuteSQLRequest DataBoostEnabled](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.ExecuteSqlRequest#Google_Cloud_Spanner_V1_ExecuteSqlRequest_DataBoostEnabled).</para>
+/// </remarks>
+public sealed class PartitionOptions
+{
+    /// <summary>
+    /// Returns a default instance of <see cref="PartitionOptions"/>.
+    /// </summary>
+    public static PartitionOptions Default { get; } = new PartitionOptions(null, null, false);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the specified new value for <see cref="DataBoostEnabled" />.
+    /// </summary>
+    /// <returns>A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithDataBoostEnabled(bool? dataBoostEnabled) => new(PartitionSizeBytes, MaxPartitions, dataBoostEnabled);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the specified new value for <see cref="PartitionSizeBytes"/>.
+    /// </summary>
+    /// <returns>A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithPartitionSizeBytes(long? partitionSizeBytes) => new(partitionSizeBytes, MaxPartitions, DataBoostEnabled);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the specified new value for <see cref="MaxPartitions"/>.
+    /// </summary>
+    /// <returns>A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithMaxPartitions(long? maxPartitions) => new(PartitionSizeBytes, maxPartitions, DataBoostEnabled);
+
+    /// <summary>
+    /// The desired data size for each partition generated.
+    /// The default value for this is null, in which case the server will decide the value of
+    /// [PartitionSizeBytes](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.PartitionOptions#Google_Cloud_Spanner_V1_PartitionOptions_PartitionSizeBytes).
+    /// </summary>
+    public long? PartitionSizeBytes { get; }
+
+    /// <summary>
+    /// The desired maximum number of partitions to return.
+    /// The default value for this is null, in which case the server will decide the value of
+    /// [MaxPartitions](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Spanner.V1/latest/Google.Cloud.Spanner.V1.PartitionOptions#Google_Cloud_Spanner_V1_PartitionOptions_MaxPartitions).
```

Comment:
Again, use `<see>` to refer to other C# code - but I don't think we need a reference at all.

Just:

> The default value for this is null, in which case the server will decide how many partitions to return.

... would be clearer IMO. Likewise for PartitionSizeBytes.